File: frontEnd/Systetica_App/lib/model/UsuarioDTO.dart
import 'package:systetica/model/RoleDTO.dart';

class UsuarioDTO {
  UsuarioDTO({
    this.id,
    this.nome,
    this.telefone,
    this.email,
    this.password,
    this.codigoAleatorio,
    this.roles,
    this.imagemBase64,
  });

  int? id;
  String? nome;
  String? telefone;
  String? email;
  String? password;
  int? codigoAleatorio;
  List<RoleDTO>? roles;
  String? imagemBase64;

  UsuarioDTO.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    nome = json['nome'];
    telefone = json['telefone'];
    email = json['email'];
    password = json['password'];
    codigoAleatorio = json['codigo_aleatorio'];
    roles = json['roles'] != null
        ? (json['roles'].map<RoleDTO>((e) => RoleDTO.fromJson(e))).toList()
        : [];
    imagemBase64 = json['imagemBase64'];
  }

  Map<String, dynamic> toJson() {
    final _data = <String, dynamic>{};
    _data['id'] = id ?? "";
    _data['nome'] = nome;
    _data['telefone'] = telefone;
    _data['email'] = email;
    _data['password'] = password;
    _data['password'] = password;
    _data['codigo_aleatorio'] = codigoAleatorio;
    _data['roles'] =
        roles != null ? roles!.map((e) => e.toString()).toList() : null;
    _data['imagemBase64'] = imagemBase64;
    return _data;
  }
}

File: backEnd/Systetica_Api/src/main/java/com/frfs/systetica/dto/UsuarioDTO.java
package com.frfs.systetica.dto;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.Column;
import java.io.Serializable;
import java.util.Collection;
import java.util.Date;
import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@JsonInclude(JsonInclude.Include.NON_NULL)
public class UsuarioDTO implements Serializable {

    @JsonProperty("id")
    private Long id;

    @JsonProperty("nome")
    private String nome;

    @JsonProperty("telefone")
    private String telefone;

    @JsonProperty("email")
    private String email;

    @JsonProperty("password")
    private String password;

    @JsonProperty("codigo_aleatorio")
    private Integer codigoAleatorio;

    @JsonProperty("data_codigo")
    private Date dataCodigo;

    @JsonProperty("usuario_ativo")
    private Boolean usuarioAtivo = false;

    @JsonProperty("data_cadastro")
    private Date dataCadastro;

    @JsonProperty("roles")
    private Collection<RoleDTO> roles;

    @JsonProperty("imagem_base64")
    private String imagemBase64;

    @JsonIgnore
    @JsonProperty("empresas")
    private List<EmpresaDTO> empresas;

    @JsonProperty("permissao_funcionario")
    private Boolean permissaoFuncionario;

    @JsonProperty("email_administrativo")
    private String emailAdministrativo;
}